src/VisualStudio/Core/Def/Options/VisualStudioSettingsOptionPersister.cs

```diff
@@ -108,99 +100,136 @@ public bool TryFetch(OptionKey2 optionKey, string storageKey, out object? value)
             return false;
         }
 
-        public Optional<object?> TryReadOptionValue(OptionKey2 optionKey, string storageKey, Type storageType)
+        public Optional<object?> TryReadAndMonitorOptionValue(OptionKey2 primaryOptionKey, string primaryStorageKey, string storageKey, Type storageType, object? defaultValue)
         {
             Contract.ThrowIfNull(_settingManager);
+            ImmutableInterlocked.GetOrAdd(ref _storageKeysToMonitorForChanges, storageKey, _ => (primaryOptionKey, primaryStorageKey));
```

Comment:
📝 This is going to box the arguments and allocate a delegate on every invocation, even if the item is already in `_storageKeysToMonitorForChanges`. Consider making the delegate static and using the overload that lets you pass custom arguments to it.